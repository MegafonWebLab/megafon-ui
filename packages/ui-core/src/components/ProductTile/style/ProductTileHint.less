@import '~styles/base.less';

@b: e('.mfui-product-tile-hint');

@{b} {
    position: absolute;
    top: -21px;
    margin: auto;
    left: 0;
    right: 0;
    border-top: 2px solid @green;
    border-bottom: 2px solid @green;
    width: 86%;

    @media @mobileB {
        width: 72%;
        top: -30px;
    }

    @media @desktopMU {
        width: 77.5%;
    }

    &__text {
        position: relative;
        padding: 9px 11px;
        box-sizing: border-box;
        background-color: @clearWhite;
        font-weight: 500;
        color: @greenLight;
        text-align: center;
        z-index: 1;

        @media @mobileB {
            padding: 9px 0;
        }
    }

    &__border {

        &:after,
        &:before {
            content: '';
            position: absolute;
            border-radius: 1px;
            top: 6px;
            width: 30px;
            height: 30px;
            transform: rotate(45deg);
            background-color: white;
            box-shadow: 0 0 0 2px @green;

            @media @mobileB {
                width: 46px;
                height: 48px;
                top: 9px;
            }
        }

        &:after {
            right: -15px;
            border-top-right-radius: 10px;

            @media @mobileB {
                right: -23px;
            }
        }

        &:before {
            left: -15px;
            border-bottom-left-radius: 10px;

            @media @mobileB {
                left: -23px;
            }
        }
    }

    &__description {
        display: inline-block;

        @media @mobileB {
            max-width: 190px;
        }
    }
}
