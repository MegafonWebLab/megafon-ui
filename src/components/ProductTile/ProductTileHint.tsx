import * as React from 'react';
import * as PropTypes from 'prop-types';
import { cnCreate } from '../../utils/cn';
import './style/ProductTileHint.less';
import { TLinkTargetType } from './ProductTile';

interface IProductTileHintProps {
    /** Title */
    title: string;
    /** Link href */
    linkHref?: string;
    /** Link target */
    linkTarget?: TLinkTargetType;
}

interface IElementAtributes {
    /** Href */
    href?: string;
    /** Target */
    target?: string;
}

const cn = cnCreate('mfui-product-tile-hint');
class ProductTileHint extends React.Component<IProductTileHintProps> {
    static propTypes = {
        linkHref: PropTypes.string,
        title: PropTypes.string,
        linkTarget: PropTypes.oneOf(['_self', '_blank', '_parent', '_top']),
    };

    static defaultProps = {
        linkTarget: '_self',
    };

    render() {
        const { title, linkHref, linkTarget } = this.props;
        const ElementType = linkHref ? 'a' : 'div';
        const attributes: IElementAtributes = linkHref ? { href: linkHref, target: linkTarget } : {};

        return (
            <div className={cn('')}>
                <ElementType {...attributes}>
                    <div className={cn('text')}>
                        <span className={cn('description')}>{title}</span>
                    </div>
                    <div className={cn('border')} />
                </ElementType>
            </div >
        );
    }
}

export default ProductTileHint;
