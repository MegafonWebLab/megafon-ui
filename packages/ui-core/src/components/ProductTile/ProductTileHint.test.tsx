import * as React from 'react';
import { shallow } from 'enzyme';
import ProductTileHint from './ProductTileHint';

const props = {
    title: 'sdfsdf',
};

describe('<ProductTileHint />', () => {
    it('it renders ProductTileHint', () => {
        const wrapper = shallow(<ProductTileHint {...props} />);
        expect(wrapper).toMatchSnapshot();
    });
});
