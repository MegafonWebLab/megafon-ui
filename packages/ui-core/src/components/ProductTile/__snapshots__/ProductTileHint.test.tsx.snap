// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ProductTileHint /> it renders ProductTileHint 1`] = `
<div
  className="mfui-product-tile-hint"
>
  <div>
    <div
      className="mfui-product-tile-hint__text"
    >
      <span
        className="mfui-product-tile-hint__description"
      >
        sdfsdf
      </span>
    </div>
    <div
      className="mfui-product-tile-hint__border"
    />
  </div>
</div>
`;
